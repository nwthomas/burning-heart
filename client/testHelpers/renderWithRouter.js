import React from "react";
import { NativeRouter as Router } from "react-router-native";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};

export default renderWithRouter;
