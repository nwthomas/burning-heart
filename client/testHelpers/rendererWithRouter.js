import React from "react";
import { NativeRouter as Router } from "react-router-native";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(<Router history={history}>{ui}</Router>),
    history
  };
};

export default rendererWithRouter;
