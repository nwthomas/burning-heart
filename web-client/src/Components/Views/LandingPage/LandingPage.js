import React, { Component } from "react";

import { CtaBanner } from "../../Presentational/CtaBanner";

import volcanoPicture from "../../../images/landing-page-image.jpg";
import fireGif from "../../../images/fire.gif";
import blankIphone from "../../../images/blank-iphone.png";
import facebookLogo from "../../../images/facebook.svg";
import instagramLogo from "../../../images/instagram.svg";
import twitterLogo from "../../../images/twitter.svg";
import linkedInLogo from "../../../images/linkedin.svg";
import closeNavbarIcon from "../../../images/menu-icon-close.svg";
import openNavbarIcon from "../../../images/menu-icon-open.svg";

class LandingPage extends Component {
  state = {
    navbarOpen: false
  };
  componentDidMount() {
    document.addEventListener("scroll", this.scrollChange);
  }
  componentWillUnmount() {
    document.removeEventListener("scroll", this.scrollChange);
  }
  handleNavbar = e => {
    e.preventDefault();
    this.setState(prevState => ({
      navbarOpen: !prevState.navbarOpen
    }));
    const body = document.querySelector("body");
    body.classList.toggle("stopScroll");
    if (!this.state.navbarOpen) {
      return (document.ontouchmove = e => e.preventDefault());
    } else {
      return (document.ontouchmove = e => true);
    }
  };
  render() {
    return (
      <>
        <CtaBanner />
        <header className="landing-page__header">
          <div className="landing-page__header__bg" />
          <nav
            className={
              this.state.navbarOpen
                ? "header__navbar header__navbar--open"
                : "header__navbar"
            }
          >
            <h1
              className={
                this.state.navbarOpen
                  ? "burning-heart__name burning-heart__name--open"
                  : "burning-heart__name"
              }
            >
              Burning Heart
            </h1>
            <div
              onClick={this.handleNavbar}
              className={
                this.state.navbarOpen
                  ? "navbar__hamburger navbar__hamburger--open"
                  : "navbar__hamburger"
              }
            >
              <img
                src={this.state.navbarOpen ? closeNavbarIcon : openNavbarIcon}
                alt="Navbar__menu__icon"
              />
            </div>
            <div
              className={
                this.state.navbarOpen
                  ? "navbar__links navbar__links--open"
                  : "navbar__links"
              }
            >
              <a href="#">Home</a>
              <a href="#">About</a>
              <a href="#">Login</a>
              <a href="#">Sign Up</a>
            </div>
          </nav>
          <div className="header__banner">
            <div className="header__image">
              <img src={volcanoPicture} alt="Smoking volcano photograph" />
            </div>
            <div className="header__content">
              <h2>For Those Who Like to Give</h2>
              <p>Burning Heart lets you donate what you want, when you want.</p>
              <p>Do what feels good. Do what makes you happy. Give.</p>
              <button className="header__button">Download</button>
            </div>
          </div>
        </header>
        <section className="middle__content">
          <div className="mobile__bg" />
          <div className="demo__information">
            <div className="demo__content">
              <h2 className="demo__header">A new way to give.</h2>
              <p className="demo__text">
                People are busy. We get it. It's hard to make time to help
                others. Burning Heart makes it easy by putting the power of
                micro-donations in the palm of your hand.
              </p>
            </div>
            <div className="iphone__picture">
              <img src={blankIphone} alt="iPhone demo of Burning Heart" />
              <img className="fire-gif" src={fireGif} alt="Fire logo" />
            </div>
          </div>
        </section>
        <footer className="footer__content">
          <div className="logo__address">
            <div className="footer__logo-name">
              <h2>Burning Heart</h2>
            </div>
            <address>
              <p>1 (800) 843-2910</p>
              <p>1400 Oakwood Avenue</p>
              <p>San Francisco, CA</p>
              <p>94574</p>
            </address>
          </div>
          <div className="footer__social__navbar">
            <nav className="footer__navbar__links">
              <a href="#">Home</a>
              <a href="#">About</a>
              <a href="#">Login</a>
              <a href="#">Sign Up</a>
            </nav>
            <div className="footer__social__links">
              <div className="social__link">
                <a href="https://www.facebook.com/nwthomas">
                  <img src={facebookLogo} alt="Facebook logo" />
                </a>
              </div>
              <div className="social__link">
                <a href="https://www.instagram.com/nwthomas/">
                  <img src={instagramLogo} alt="Instagram logo" />
                </a>
              </div>
              <div className="social__link">
                <a href="https://twitter.com/nwthomas_">
                  <img src={twitterLogo} alt="Twitter logo" />
                </a>
              </div>
              <div className="social__link">
                <a href="https://www.linkedin.com/in/nwthomas-profile/">
                  <img src={linkedInLogo} alt="LinkedIn logo" />
                </a>
              </div>
            </div>
          </div>
        </footer>
      </>
    );
  }
}

export default LandingPage;
