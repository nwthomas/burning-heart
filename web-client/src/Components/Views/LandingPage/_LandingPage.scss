.landing-page__header {
  width: 100%;
  position: relative;
  margin: 0 auto;
  max-width: 1600px;

  .landing-page__header__bg {
    position: absolute;
    z-index: -1;
    top: 0;
    right: 0;
    width: 60%;
    height: 70vw;
    max-height: 950px;
    background-image: linear-gradient(to right, #3955f2, #5d73f4);

    @media #{$desktop} {
      height: 90vw;
    }

    @media #{$tablet} {
      height: 105vw;
      width: 100%;
      min-height: 750px;
    }

    @media #{$mobile} {
      height: 105vw;
      min-height: 650px;
    }
  }

  .header__navbar {
    width: 100%;
    height: 80px;
    padding: 20px 3% 0;
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media #{$mobile} {
      flex-direction: row;
      align-content: flex-start;
      flex-wrap: wrap;
    }

    .burning-heart__name {
      font-size: 3.8rem;
      font-weight: 700;
      color: #ffb2b3;

      @media #{$desktop} {
        font-size: 3rem;
      }

      @media #{$mobile} {
        font-size: 2.4rem;
        align-items: flex-start;
        width: 70%;
      }
    }

    .burning-heart__name--open {
      margin-top: 10px;
    }

    .navbar__hamburger {
      display: none;

      @media #{$mobile} {
        display: flex;
        width: 40px;
        height: 40px;
        margin-top: 1px;
        cursor: pointer;

        &:hover {
          opacity: 0.8;
        }
      }
    }

    .navbar__hamburger--open {
      margin-top: 3px;
    }

    .navbar__links {
      width: 55%;
      max-width: 1000px;
      display: flex;
      justify-content: flex-end;
      margin-top: 10px;

      @media #{$mobile} {
        display: none;
      }

      a {
        width: 20%;
        text-align: right;
        color: white;
        font-size: 1.6rem;
        font-weight: 700;

        @media #{$desktop} {
          width: 30%;
        }

        @media #{$mobile} {
          text-align: left;
        }
      }
    }

    .navbar__links--open {
      display: flex;
      flex-direction: column;
      width: 30%;

      a {
        margin-top: 20px;
        width: 100%;
      }
    }
  }

  .header__navbar--open {
    height: 100vh;
    width: 100%;
    padding-top: 19px;
    background: black;
    align-items: flex-start;
    position: fixed;
    z-index: 50;
  }

  .header__banner {
    display: flex;
    justify-content: flex-start;

    @media #{$tablet} {
      flex-direction: column-reverse;
    }

    .header__image {
      z-index: 15;
      margin-top: 100px;
      width: 50%;
      max-width: 1100px;

      @media #{$tablet} {
        width: 70%;
        align-self: flex-end;
      }

      @media #{$mobile} {
        width: 90%;
      }
    }

    .header__content {
      margin: 100px 0 0 3%;
      max-width: 40%;

      @media #{$tablet} {
        max-width: 80%;
      }

      @media #{$tablet} {
        margin-top: 50px;
      }

      h2 {
        font-size: 5rem;
        color: white;
        font-weight: 700;

        @media #{$desktop} {
          font-size: 4rem;
        }

        @media #{$mobile} {
          font-size: 3rem;
        }
      }

      p {
        font-family: "Montserrat", sans-serif;
        font-size: 2.4rem;
        color: white;
        font-weight: 700;
        line-height: 1.5;
        margin-top: 50px;

        @media #{$max-desktop} {
          margin-top: 30px;
          font-size: 2rem;
        }

        @media #{$mobile} {
          font-size: 1.6rem;
        }
      }

      button {
        margin-top: 50px;
        height: 50px;
        width: 200px;
        border-radius: 50px;
        background: #ffb2b3;
        color: white;
        font-weight: 700;
        font-size: 1.6rem;
        border: 1px solid #ffb2b3;
        cursor: pointer;

        &:hover {
          color: #ffb2b3;
          background: white;
        }
      }
    }
  }
}

.middle__content {
  width: 100%;
  margin: 0 auto;
  max-width: 1600px;
  padding-bottom: 100px;

  @media #{$tablet} {
    position: relative;

    .mobile__bg {
      position: absolute;
      bottom: 0;
      width: 100%;
      height: 450px;
      background: linear-gradient(to right, #ffff51, #ffffa0);
      z-index: -1;
    }
  }

  .demo__information {
    width: 100%;
    padding: 0 3%;
    display: flex;
    justify-content: space-between;

    @media #{$tablet} {
      display: flex;
      flex-direction: column;
    }

    .demo__content {
      margin-top: 240px;
      width: 45%;
      font-weight: 700;
      color: #ffb2b3;

      @media #{$desktop} {
        margin-top: 350px;
      }

      @media #{$tablet} {
        margin-top: 90px;
        width: 100%;
      }

      .demo__header {
        font-size: 5rem;

        @media #{$max-desktop} {
          font-size: 4rem;
        }

        @media #{$tablet} {
          text-align: center;
        }

        @media #{$mobile} {
          font-size: 3rem;
        }
      }

      .demo__text {
        line-height: 1.5;
        margin-top: 50px;
        font-family: "Montserrat", sans-serif;
        font-size: 2.4rem;

        @media #{$max-desktop} {
          margin-top: 30px;
          font-size: 2rem;
        }

        @media #{$tablet} {
          text-align: center;
          padding: 0 5%;
        }

        @media #{$mobile} {
          font-size: 1.6rem;
        }
      }
    }

    .iphone__picture {
      width: 280px;
      margin: 100px 15% 0 0;
      position: relative;
      display: flex;
      justify-content: center;
      align-items: center;

      @media #{$max-desktop} {
        margin: 70px 13% 0 0;
      }

      @media #{$tablet} {
        margin: 100px auto 0;
      }

      .fire-gif {
        width: 40%;
        position: absolute;
      }
    }
  }
}

.footer__content {
  background-image: linear-gradient(to right, #f7f8f9, #e9eeef);
  width: 100%;
  max-width: 1600px;
  padding: 82px 3% 62px;
  margin: 0 auto 0;
  display: flex;
  justify-content: space-between;

  @media #{$tablet} {
    flex-direction: row;
  }

  @media #{$mobile} {
    flex-direction: column;
    padding-bottom: 72px;
  }

  .logo__address {
    .footer__logo-name {
      height: 70px;
      width: 320px;
      background: black;

      @media #{$tablet} {
        width: 260px;
        height: 60px;
      }

      @media #{$mobile} {
        margin-bottom: 50px;
      }

      h2 {
        line-height: 70px;
        width: 100%;
        text-align: center;
        font-size: 3.8rem;
        font-weight: 700;
        color: white;

        @media #{$tablet} {
          font-size: 3rem;
          line-height: 60px;
        }
      }
    }

    address {
      p {
        font-weight: 700;
        margin-top: 20px;
      }
    }
  }

  .footer__social__navbar {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-end;
    width: 55%;

    @media #{$tablet} {
      flex-direction: row;
      align-items: flex-start;
    }

    @media #{$mobile} {
      flex-direction: column;
      width: 100%;
    }

    .footer__navbar__links {
      display: flex;
      justify-content: flex-end;
      width: 100%;
      max-width: 600px;

      @media #{$tablet} {
        max-width: 800px;
        flex-direction: column;
        margin-top: 80px;
      }

      @media #{$mobile} {
        margin-top: 50px;
      }

      a {
        text-align: right;
        width: 20%;
        color: black;
        font-size: 1.6rem;
        font-weight: 700;

        @media #{$tablet} {
          width: 70px;
          text-align: left;
          margin: 0 15% 20px;
        }

        @media #{$mobile} {
          margin: 0 0 20px;
        }
      }
    }

    .footer__social__links {
      display: flex;
      justify-content: flex-end;
      width: 100%;
      margin-top: 120px;

      @media #{$tablet} {
        flex-direction: column;
        justify-content: flex-start;
        align-items: flex-end;
        margin-top: 0;
        width: 25%;
      }

      @media #{$mobile} {
        flex-direction: row;
        width: 100%;
      }

      .social__link {
        width: 50px;
        height: auto;
        margin: 20px 0 0 6%;

        @media #{$tablet} {
          margin: 0 0 8px;
          width: 40px;
        }

        @media #{$mobile} {
          margin: 30px 5% 0 0;
        }

        &:hover {
          opacity: 0.7;
        }
      }
    }
  }
}
