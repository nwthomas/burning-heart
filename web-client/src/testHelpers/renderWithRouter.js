import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default renderWithRouter;
