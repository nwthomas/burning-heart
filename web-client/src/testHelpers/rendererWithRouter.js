import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default rendererWithRouter;
