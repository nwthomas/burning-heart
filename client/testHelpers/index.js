import renderWithRouter from "./renderWithRouter";
import rendererWithRouter from "./rendererWithRouter";

export { renderWithRouter, rendererWithRouter };
