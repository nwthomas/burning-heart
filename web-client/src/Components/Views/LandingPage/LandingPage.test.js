import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../../testHelpers";

import LandingPage from "./LandingPage";

describe("<LandingPage />", () => {
  it("should render the LandingPage component without crashing", () => {
    const helpers = renderWithRouter(<LandingPage />);
  });

  it("should match the snapshot of LandingPage", () => {
    const tree = rendererWithRouter(<LandingPage />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
