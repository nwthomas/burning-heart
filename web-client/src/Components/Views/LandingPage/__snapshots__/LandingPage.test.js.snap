// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> should match the snapshot of LandingPage 1`] = `
Array [
  <div
    className="cta__banner"
  >
    <p
      className="cta__banner__text"
    >
      Burning Heart is a donation app for people who like to give what they can, when they can.
       
      <a
        className="cta__banner__link"
        href="/signup"
        onClick={[Function]}
      >
        Sign up here.
      </a>
    </p>
  </div>,
  <header
    className="landing-page__header"
  >
    <div
      className="landing-page__header__bg"
    />
    <nav
      className="header__navbar"
    >
      <a
        className="burning-heart__name"
        href="/"
        onClick={[Function]}
      >
        Burning Heart
      </a>
      <div
        className="navbar__hamburger"
        onClick={[Function]}
      >
        <img
          alt="navbar__menu__icon"
          src="menu-icon-open.svg"
        />
      </div>
      <div
        className="navbar__links"
      >
        <a
          aria-current="page"
          className="navbar__link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="navbar__link"
          href="/home"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          aria-current={null}
          className="navbar__link"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </nav>
    <div
      className="header__banner"
    >
      <div
        className="header__image"
      >
        <img
          alt="Smoking volcano photograph"
          src="landing-page-image.jpg"
        />
      </div>
      <div
        className="header__content"
      >
        <h2>
          For Those Who Like to Give
        </h2>
        <p>
          Burning Heart lets you donate what you want, when you want.
        </p>
        <p>
          Do what feels good. Do what makes you happy. Give.
        </p>
        <button
          className="header__button"
        >
          Download
        </button>
      </div>
    </div>
  </header>,
  <section
    className="middle__content"
  >
    <div
      className="mobile__bg"
    />
    <div
      className="demo__information"
    >
      <div
        className="demo__content"
      >
        <h2
          className="demo__header"
        >
          A new way to give.
        </h2>
        <p
          className="demo__text"
        >
          People are busy. We get it. It's hard to make time to help others. Burning Heart makes it easy by putting the power of micro-donations in the palm of your hand.
        </p>
      </div>
      <div
        className="iphone__picture"
      >
        <img
          alt="iPhone demo of Burning Heart"
          src="blank-iphone.png"
        />
        <img
          alt="Fire logo"
          className="fire-gif"
          src="fire.gif"
        />
      </div>
    </div>
  </section>,
  <footer
    className="footer__content"
  >
    <div
      className="logo__address"
    >
      <div
        className="footer__logo-name"
      >
        <h2>
          Burning Heart
        </h2>
      </div>
      <address>
        <p>
          1 (800) 843-2910
        </p>
        <p>
          1400 Oakwood Avenue
        </p>
        <p>
          San Francisco, CA
        </p>
        <p>
          94574
        </p>
      </address>
    </div>
    <div
      className="footer__social__navbar"
    >
      <nav
        className="footer__navbar__links"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
        <a
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </nav>
      <div
        className="footer__social__links"
      >
        <div
          className="social__link"
        >
          <a
            href="https://www.facebook.com/nwthomas"
          >
            <img
              alt="Facebook logo"
              src="facebook.svg"
            />
          </a>
        </div>
        <div
          className="social__link"
        >
          <a
            href="https://www.instagram.com/nwthomas/"
          >
            <img
              alt="Instagram logo"
              src="instagram.svg"
            />
          </a>
        </div>
        <div
          className="social__link"
        >
          <a
            href="https://twitter.com/nwthomas_"
          >
            <img
              alt="Twitter logo"
              src="twitter.svg"
            />
          </a>
        </div>
        <div
          className="social__link"
        >
          <a
            href="https://www.linkedin.com/in/nwthomas-profile/"
          >
            <img
              alt="LinkedIn logo"
              src="linkedin.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
